src/errors.md

```diff
@@ -0,0 +1,128 @@
+(Caused by NewConnectionError('<urllib3.connection.HTTPConnection object at 0x109f80c88>: Failed to establish a new connection: [Errno 61] Connection refused',))
+```
+
+This error occurs when you make more than X login calls in a given minute. You just need to wait a bit for it to disappear. The value of X depends on which value you set for the `DEFAULT_THROTTLE_RATES` environment variable. Default value when launching with skaffold is 120. (tips from [jmorel](https://github.com/jmorel) [here](https://github.com/SubstraFoundation/substra/issues/209))
+</details>
+
+### substra list/get/describe/download an asset
+
+<details>
+<summary><b>HTTPConnectionPool</b></summary>
+
+```sh
+substra list dataset
+
+Error: Request failed: ConnectionError: HTTPConnectionPool(host='127.0.0.1', port=8000): Max retries exceeded with url: /data_manager/ (Caused by NewConnectionError('<urllib3.connection.HTTPConnection object at 0x7f4ec7b4a910>: Failed to establish a new connection: [Errno 111] Connection refused'))
+```
+
+This error is related to the fact that you need to provide a `--profile` option: `substra list dataset --profile node-1`.
+</details>
+
+<details>
+<summary><b>Invalid token</b></summary>
+
+```sh
+substra list dataset --profile node-1
+
+Requests error status 401: {"detail":"Invalid token."}
+Error: Request failed: AuthenticationError: 401 Client Error: Unauthorized for url: http://substra-backend.node-1.com/data_manager/
+```
+
+Please login again in order to refresh the token.
+
+> Note: Please be aware that initiating a new login will logout any already active connection!
+
+</details>
+
+<details>
+<summary><b>No asset for key <KEY></b></summary>
+
+```sh
+get dataset 09c741742ec4ce10360e0dcd1ca0b8ecf8edf97263593d5f6b1ce6c657d54c8c --profile node-1
+Requests error status 404: {"message":"no asset for key 09c741742ec4ce10360e0dcd1ca0b8ecf8edf97263593d5f6b1ce6c657d54c8c"}
+Error: Request failed: NotFound: 404 Client Error: Not Found for url: http://substra-backend.node-1.com/data_manager/09c741742ec4ce10360e0dcd1ca0b8ecf8edf97263593d5f6b1ce6c657d54c8c/
+```
+
+Please check the item key you are providing (with `substra list <ASSET>`).
+
+</details>
+
+## Python SDK
+
+[SDK documentation](https://github.com/SubstraFoundation/substra/blob/master/references/sdk.md#substrasdk)
+
+> Remember: debug mode is awesome, use it! You just need to define your client like this `client = substra.Client(debug=True)`
+> to be able to use `pdb`/`ipdb`!
+
+### Generate data sample
+
+Check the `generate_data_samples.py` script and the path to your dataset, the values of `N_TRAIN_DATA_SAMPLES` and `N_TEST_DATA_SAMPLES`.
+
+<details>
+<summary><b>FileExistsError</b></summary>
+This indicates that the `train_data_samples` & `test_data_samples` have already been generated in the `assets` folder. This means that you will need to remove it before re-generating data samples.
+</details>
+
```

Comment:
I believe this is specific to the titanic example? If so I would remove it. Or maybe put it in the readme of the example. What do you think @natct10?

Reply:
It is indeed based on Titanic example but I though it was a good illustration of the `FileExistsError` error. Are you sure to want to remove it?

Reply:
`FileExistsError` is a python error but unrelated to Substra?
It appends there --> https://github.com/SubstraFoundation/substra/blob/master/examples/titanic/scripts/generate_data_samples.py#L66 ?

Reply:
Yep this is indeed a python error you can face when you have already generated samples. I can remove this if you prefer